### PR TITLE
feat(database-settings): add resources section

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

[> Link to the JIRA ticket
](https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-520)

- Add section Resources for Database
- Add a shared component to manage Units (add it for the Resources General section too)

Capture d’écran 2022-09-05 à 15 52 59

---

### Global

- [x] This PR does not introduce any breaking change
- [ ] This PR introduces breaking change(s) and has been labeled as such
- [x] I have found someone to review this PR and pinged him

### Store

- [ ] This PR introduces new store changes

### NX

- [x] I have run the dep-graph locally and made sure the tree was clean i.e no circular dependencies
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`

### Clean Code

- [x] I made sure the code is type safe (no any)
- [ ] I have included a feature flag on my feature, if applicable

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.  
Click on the link below to follow its deployment and use it.  
:point_right: [[PR] staging - feat(database-settings): add resources section - 2022-09-02T08:03:40Z](https://console.qovery.com/platform/organization/141c07c8-0dd9-4623-983b-3fdd61867255/projects/977bd069-20f7-4b24-bd95-28b44174cf43/environments/dddd75b1-281b-48b2-bddc-356c57ebfd07/applications)

**Review by bdebon — APPROVED** [review]
I was suprised at the beginning to see a form component that embeds the `Controller` directly. This component is then not usable is standalone mode. But in the same way, it's such a specific component that we will always use it in a form context.
This component is now completely dependant of `react-hook-form` which I think is not a good practice but for this very specific component let's say it's ok.

File: libs/shared/ui/src/lib/components/inputs/input-size-unit/input-size-unit.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  const [memorySize, setMemorySize] = useState<string | MemorySizeEnum>(currentUnit || MemorySizeEnum.MB)
+
+  const handleChangeMemoryUnit = (size: string | MemorySizeEnum) => {
+    getUnit && getUnit(size)
+    setMemorySize(size)
+
+    if (size !== memorySize) {
+      const currentSizeByUnit = getSizeUnit(size, value)
+      setValue && setValue(name, currentSizeByUnit)
```
Comment (bdebon) [inline libs/shared/ui/src/lib/components/inputs/input-size-unit/input-size-unit.tsx:40]: Remi, I'm not convinced that you need to give the setValue, you can directly do that with the onChange I think.
I mean, you can call onChange({name, currentSizeByUnit})

It makes this component even less complex to use. In the onChange you will have your `field.onChange` that does the same as setValue(name, value)
Do you see what I mean?

**Pull request merged**